src/ProjectOrigin.Electricity.IntegrationTests/Helpers/ElectricityClientTestBase.cs

```diff
@@ -0,0 +1,44 @@
+using ProjectOrigin.Electricity.Client;
+using ProjectOrigin.Electricity.Client.Models;
+using ProjectOrigin.Electricity.Server;
+using Xunit.Abstractions;
+
+namespace ProjectOrigin.Electricity.IntegrationTests.Helpers
+{
+    public class ElectricityClientTestBase : IntegrationTestBase, IDisposable
+    {
+        private ElectricityClient? _client;
+        private SemaphoreSlim _semaphore = new SemaphoreSlim(0, 1);
```

Comment:
For the uninitiated like myself the SemaphoreSlim is a lightweight threading alternative that limits the number of threads that can access a resource or pool of resources concurrently (https://learn.microsoft.com/en-us/dotnet/api/system.threading.semaphoreslim?view=net-7.0)[https://learn.microsoft.com/en-us/dotnet/api/system.threading.semaphoreslim?view=net-7.0]